akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtoAnySerialization.scala

```diff
@@ -167,16 +175,62 @@ import scalapb.options.Scalapb
         else
           idAndManifest.substring(0, i).toInt -> idAndManifest.substring(i + 1)
 
+      println(s"# deserialize 3: $typeUrl, akka $id $manifest") // FIXME
+
       serialization.deserialize(scalaPbAny.value.toByteArray, id, manifest).get
     } else if (prefer == Prefer.Scala) {
+      println(s"# deserialize 4: $typeUrl") // FIXME
       // when custom typeUrl
       scalaPbAny
     } else {
+      println(s"# deserialize 5: $typeUrl") // FIXME
       // when custom typeUrl
       ScalaPbAny.toJavaProto(scalaPbAny)
     }
   }
 
+  def toSerializedEvent(scalaPbAny: ScalaPbAny): Option[SerializedEvent] = {
+    // see corresponding typeUrl cases in `deserialize`
+
+    def pbAnyToSerializedEvent(): SerializedEvent = {
+      val pbAny =
+        if (prefer == Prefer.Scala) scalaPbAny
+        else ScalaPbAny.toJavaProto(scalaPbAny)
+      val serializer = serialization.findSerializerFor(pbAny)
+      val manifest = Serializers.manifestFor(serializer, pbAny)
+      SerializedEvent(scalaPbAny.toByteArray, serializer.identifier, manifest)
```

Comment:
Note that this is the toByteArray of the PbAny. I think that corresponds to what Akka's ProtobufSerializer would do in toBinary. Maybe it is not safe to assume that any serializer would produce the same bytes as this. Maybe best to not use the SerializedEvent for this case and then fall back to the ordinary deserialize.